### ISSUE TITLE
bug in unify_with_occurs_check/2

### ISSUE THREAD

**Issue by jburse** [SWI-Prolog/swipl-devel#758]
Smallest test case I could produce, not found on SWI-Prolog discourse:
```
?- freeze(A,(write(A),nl)),freeze(B,(write(B),nl)),unify_with_occurs_check(B-A,n-s(A)).
false.

?- freeze(A,(write(A),nl)),freeze(B,(write(B),nl)),unify_with_occurs_check(A-B,s(A)-n).
@(S_1,[S_1=s(S_1)])
n
A = s(A),
B = n.
```
The outcome of both unifications with occurs check should be false. 
But the second query delivers a success.

**Comment by JanWielemaker** [comment]
Thanks.  Small programs help.  Fixed with 391652e65fab696df9ef3a6d0f0a7862c857c683.  Also fixes the [quine](https://swi-prolog.discourse.group/t/is-there-a-whened-2-predicate/3261/18?u=jan) issue.  Now runs in 7.7 sec.

**Comment by JanWielemaker** [comment]
This issue has been mentioned on **SWI-Prolog**. There might be relevant details there:

https://swi-prolog.discourse.group/t/is-there-a-whened-2-predicate/3261/22

**Comment by jburse** [comment]
So the bug has been fixed in SWI-Prolog 8.3.17 (devel) and
was backported to SWI-Prolog 8.2.4 (stable) as well.

Cool, thanks!
